cdisc_rules_engine/utilities/sql_rule_processor.py

```diff
@@ -447,14 +447,14 @@ def is_suitable_for_validation(
         rule_id = rule.get("core_id", "unknown")
         dataset_name = dataset_metadata.dataset_name
 
+        if not self.valid_rule_structure(rule):
```

Comment:
This is definitely the wrong place for this check. It should happen way, way earlier. Why is it getting to this point if it's an invalid rule?